### DISCUSSION TITLE
cant find gpio0

### DISCUSSION THREAD

**Discussion by lukasa1993** [arendst/Tasmota#10672] [Q&A] [answered]
this is some kind sonoff switch 1 gang EU software is ITEAD probably new one tuya convert didn't work

![image](https://user-images.githubusercontent.com/1499012/105575613-55b72980-5d86-11eb-80b6-f516b9f12a6f.jpeg)

![image](https://user-images.githubusercontent.com/1499012/105575618-5d76ce00-5d86-11eb-813d-1d071568e90b.jpeg)

**Comment by lalo-uy** [comment, marked as answer]
That chip AltoBem is not ESP, nor Realtek. Tasmota is not compatible with
this one.

**Reply by lukasa1993** [reply]
> That chip AltoBem is not ESP, nor Realtek. Tasmoto is not compatible with this one. El El sáb, 23 de ene. de 2021 a la(s) 07:23, luka dodelia < notifications@github.com> escribió:
> […](#)

are you sure? i bought it as esp8266

**Comment by meingraham** [comment]
> are you sure? i bought it as esp8266

What, the product description was wrong? Say it isn't so! 😉 

[AltoBeam 6431](https://www.altobeam.com/en/contents/73/10.html) <> ESP82xx

You can only use Tuya Convert on devices manufactured by Tuya... and then only if the device uses an ESP82xx.

This is a KingArt device, not Sonoff and therefore likely does not use the ITEAD firmware. If it were, then the device would be ESP82xx based. I'm not aware of ITEAD supporting anything but the ESP82xx at this time.

**Reply by lukasa1993** [reply]
> > are you sure? i bought it as esp8266
> 
> What, the product description was wrong? Say it isn't so! 😉
> 
> [AltoBeam 6431](https://www.altobeam.com/en/contents/73/10.html) <> ESP82xx
> 
> You can only use Tuya Convert on devices manufactured by Tuya... and then only if the device uses an ESP82xx.
> 
> This is a KingArt device, not Sonoff and therefore likely does not use the ITEAD firmware. If it were, then the device would be ESP82xx based. I'm not aware of ITEAD supporting anything but the ESP82xx at this time.

when it boots natively it generates ITEAD-XXXX wifi ap

**Comment by meingraham** [comment]
> when it boots natively it generates ITEAD-XXXX wifi ap

This is not good news. It means that ITEAD also is moving to something other than the ESP82xx.

**Reply by lalo-uy** [reply]
No, this is a compatible device, but no build by Sonoff.
They mimic the wifi network name.

**Reply by lukasa1993** [reply]
say more how i flash it

**Reply by meingraham** [reply]
@lukasa1993

The Wi-Fi SoC inside your "compatible" switch is NOT compatible with the ESP82xx.

What @lalo-uy is saying is that your KingArt is a compatible touch light switch device similar to a device built by Sonoff. KingArt appear to have moved away from the Espressif chips to these AltoBeam chips. What it does look like is that ITEAD has started supporting other Wi-Fi SoC in addition to the ESP82xx.

Hopefully Sonoff will stick with the ESP82xx.

**Reply by ascillato** [reply]
@lukasa1993

Sorry but your device can not be flashed with Tasmota. That is an unsupported device because it has an unsupported chip.

You can use as it comes from the manufacturer or return to the seller and buy a different one.

Your device is not supported because the main controller is not an ESP8266 chip. Tasmota only works on ESP8266, ESP8285 and ESP32 chips. 

We can't add support for other chips that don't run with the Arduino Core. Sorry.

**Comment by Frogmore42** [comment]
here is some information on the chip (which is not an esp8266) https://www.altobeam.com/en/contents/73/10.html

**Reply by lukasa1993** [reply]
found that page too but not real information at all

**Comment by meingraham** [comment]
That's only to say that the chip is indeed the Wi-Fi SoC and that your device is therefore not ESP82xx based... and therefore Tasmota is not an option. Very few of these "newer" Wi-Fi SoC have any substantial amount of custom firmware work going on... so you're not likely to find anything more about them.

**Comment by ascillato** [comment]
@lukasa1993

Sorry but your device can not be flashed with Tasmota. That is an unsupported device because it has an unsupported chip.

You can use as it comes from the manufacturer or return to the seller and buy a different one.

Your device is not supported because the main controller is not an ESP8266 chip. Tasmota only works on ESP8266, ESP8285 and ESP32 chips. 

We can't add support for other chips that don't run with the Arduino Core. Sorry.

**Comment by barbudor** [comment]
@lukasa1993 Before purchasing a device, please check our repository of devices that have worked and should still work
https://templates.blakadder.com/

HOwever, manufacturer based on "Tuya" may also moved away from ESP8266 without letting us known ;) So no warranty.

If you want to be 100% sure, you can order of the devices pre-flashed with Tasmota:
https://templates.blakadder.com/preflashed.html

**Reply by lukasa1993** [reply]
umm do you know any good one which is black 1 gang ? like this one ?

![s-l1600](https://user-images.githubusercontent.com/1499012/105685685-c8074580-5f0f-11eb-9f89-016c948eac1c.jpg)
